pkg/liqo-controller-manager/namespaceMap-controller/update_namespaceoffloading_status.go

```diff
@@ -0,0 +1,150 @@
+package namespacemapctrl
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+)
+
+// If all requested remote namespaces have been created, it is necessary to set the global status of Offloading,
+// otherwise the Offloading is still in progress.
+func (r *NamespaceMapReconciler) checkNamespaceOffloadingStatus(namespaceOffloading *offv1alpha1.NamespaceOffloading) error {
+	virtualNodes := &corev1.NodeList{}
+	if err := r.List(context.TODO(), virtualNodes,
+		client.MatchingLabels{liqoconst.TypeLabel: liqoconst.TypeNode}); err != nil {
+		klog.Error(err, " --> Unable to List all virtual nodes")
+		return err
+	}
+
+	if len(virtualNodes.Items) == 0 {
+		err := fmt.Errorf(" No VirtualNode at the moment")
+		klog.Error(err)
+		return err
+	}
+
+	if len(namespaceOffloading.Status.RemoteNamespacesConditions) == len(virtualNodes.Items) {
+		ready := 0
+		notReady := 0
+
+		for i := range namespaceOffloading.Status.RemoteNamespacesConditions {
+			condition := liqoutils.GetNamespaceOffloadingCondition(
+				namespaceOffloading.Status.RemoteNamespacesConditions[i], offv1alpha1.NamespaceReady)
+			if condition == nil {
+				continue
+			}
+			if condition.Status == corev1.ConditionTrue {
+				ready++
+			} else {
+				notReady++
+			}
+		}
+
+		switch {
+		case ready == 0:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.AllFailedOffloadingPhaseType
```

Comment:
Do we need to differentiate among SomeFailed and AllFailed? What do you think @giorio94?

Reply:
I would personally differentiate, since SomeFailed means you can still perform offloading, but some name conflict arose, while AllFailed is sorta catastrophic. Names could be clearer, but I couldn't come up with better ideas at the time :(

Reply:
Makes sense 👍 thanks!